policy/release/schedule_test.rego

```diff
@@ -0,0 +1,104 @@
+package policy.release.schedule
+
+import data.lib
+
+test_no_restriction_by_default {
+	lib.assert_empty(deny)
+}
+
+test_weekday_restriction {
+	disallowed := ["friday", "saturday", "sunday"]
+
+	lib.assert_empty(deny) with data.rule_data.disallowed_weekdays as disallowed
+		with data.config.policy.when_ns as monday
+
+	lib.assert_empty(deny) with data.rule_data.disallowed_weekdays as disallowed
+		with data.config.policy.when_ns as tuesday
+
+	lib.assert_empty(deny) with data.rule_data.disallowed_weekdays as disallowed
+		with data.config.policy.when_ns as wednesday
+
+	lib.assert_empty(deny) with data.rule_data.disallowed_weekdays as disallowed
+		with data.config.policy.when_ns as thursday
+
+	friday_violation := {{
+		"code": "schedule.weekday_restriction",
+		"msg": "friday is a disallowed weekday: friday, saturday, sunday",
+	}}
+	lib.assert_equal_results(deny, friday_violation) with data.rule_data.disallowed_weekdays as disallowed
+		with data.config.policy.when_ns as friday
+
+	saturday_violation := {{
+		"code": "schedule.weekday_restriction",
+		"msg": "saturday is a disallowed weekday: friday, saturday, sunday",
+	}}
+	lib.assert_equal_results(deny, saturday_violation) with data.rule_data.disallowed_weekdays as disallowed
+		with data.config.policy.when_ns as saturday
+
+	sunday_violation := {{
+		"code": "schedule.weekday_restriction",
+		"msg": "sunday is a disallowed weekday: friday, saturday, sunday",
+	}}
+	lib.assert_equal_results(deny, sunday_violation) with data.rule_data.disallowed_weekdays as disallowed
+		with data.config.policy.when_ns as sunday
+}
+
+test_weekday_restriction_case_insensitive {
+	violation := {{
+		"code": "schedule.weekday_restriction",
+		"msg": "friday is a disallowed weekday: friday",
+	}}
+
+	lib.assert_equal_results(deny, violation) with data.rule_data.disallowed_weekdays as ["FRIDAY"]
+		with data.config.policy.when_ns as friday
+	lib.assert_empty(deny) with data.rule_data.disallowed_weekdays as ["FRIDAY"]
+		with data.config.policy.when_ns as monday
+
+	lib.assert_equal_results(deny, violation) with data.rule_data.disallowed_weekdays as ["friday"]
+		with data.config.policy.when_ns as friday
+	lib.assert_empty(deny) with data.rule_data.disallowed_weekdays as ["friday"]
+		with data.config.policy.when_ns as monday
+}
+
+test_date_restriction {
+	year_violation := {{
+		"code": "schedule.date_restriction",
+		"msg": "2023-01-01T00:00:00Z is a disallowed date: 2023",
+	}}
+	lib.assert_equal_results(deny, year_violation) with data.rule_data.disallowed_dates as ["2023"]
```

Comment:
I don't think it makes much sense to support date ranges like this, best have the dates explicitly stated in full, a typo in the configuration could lead to wide denies

Reply:
I think ranges would be useful, typos are always a risk, and you'd learn pretty quickly if every release attempt is getting blocked.

Reply:
And blocking on a 10day/month/01-09,10-12month/year range is okay? That just seems off to me. One could provide a list of dates and do arbitrary ranges instead, would that be okay?

Reply:
Oh I missed some of the context, yeah blocking a whole year doesn't seem useful, arbitrary ranges is more what I had in mind. Days of the week, weekends, arbitrary ranges

Reply:
Yeah, agree. I've changed it so now you have to use an explicit date, e.g. `2023-01-01`. Let's see how far we can get with this for now before adding support for ranges.